02-express-tutorial/answers/07-params-query.js

```diff
@@ -0,0 +1,66 @@
+const express = require("express");
+const app = express();
+const { products } = require("./data");
+
+app.get("/", (req, res) => {
+  res.send('<h1>Home page!</h1><a href="api/products">Products</a>');
+});
+
+app.get("/api/products", (req, res) => {
+  const newProducts = products.map((product) => {
+    const { id, name, image } = product;
+    return { id, name, image };
+  });
+  res.json(newProducts);
+});
+
+app.get("/api/products/1", (req, res) => {
+  const singleProduct = products.find((product) => product.id === 1);
+  res.json(singleProduct);
+});
+
+app.get("/api/products/:productID", (req, res) => {
+  //console.log(req);
+  //console.log(req.params);
+  const { productID } = req.params;
+  const singleProduct = products.find(
+    (product) => product.id === Number(productID)
+  );
+  if (!singleProduct) {
+    return res.status(404).send("Product Does Not Exist");
+  }
+  return res.json(singleProduct);
+});
+
+// this part doesn't work. I don't know why
+app.get("/api/products/:productId/reviews/:reviewID", (req, res) => {
+  console.log(req.params);
+  res.send("Hello World!");
+});
```

Comment:
This works for me when I make a request like this:

```
curl localhost:5001/api/products/1/reviews/1
```

What was failing for you?